### ISSUE TITLE
Disabling a node does not affect the simulations

### ISSUE THREAD

**Issue by BrianCollinss** [APSIMInitiative/ApsimX#8562]
### What happened?

For example:

![image](https://github.com/APSIMInitiative/ApsimX/assets/39573622/404316f6-c17a-4265-b4d4-7228662c6913)

### What Operating System are you on?

Windows

**Comment by hut104** [comment]
Is there an example simulation to demonstrate this behaviour?

**Comment by BrianCollinss** [comment]
I'm on my cellphone. I'll send the example when I have my laptop. First time I noticed this behaviour was when I was testing MinSoilTemperature under GerminatingPhase (all under Replacements). Disabling it didn't affect the results but removing it caused error. Now I'm modifying RUE modifiers and FRGR children and it's not working.

**Comment by BrianCollinss** [comment]
[Soybean.zip](https://github.com/APSIMInitiative/ApsimX/files/13999700/Soybean.zip)

You can see that, for example, disabling FW under Photosynthesis does not change the results. But removing it causes an error.

**Comment by par456** [comment]
Having a look at this, it seems that when a model is copied from Replacements, each node's enabled flag is set to the value of the model it is overwriting, not what it was set to under replacements.

In this case, FW is a required property of Photosynthesis, so if it's disabled, it should actually throw an error.

Is this working correctly @hol353? Seems like if you disable a child model in replacements, it shouldn't get re-enabled when it replaces the existing enabled model.

**Comment by BrianCollinss** [comment]
What about nodes that do not exist in the overwritten model? In my example, FW (under FRGR) does not exists in the original model.

**Comment by hol353** [comment]
I think models that exist under replacements should be exactly what you get in your simulation when it is run. If a model in replacements is disabled then it should also be disabled in the simulation when the replacement happens. If a model exists in replacements that doesn't exist in the simulation it should be added. We might need a bunch of unit tests for this.

**Comment by ric394** [comment]
@hol353 handling replacements in that way makes quickly disabling replacements impossible. Did we want to create an option to disable children in replacements if we go forward with this new way of handling replacements?

**Comment by par456** [comment]
We've looked at how this is working and it is currently working as intended, however we think it should be changed to make a bit more sense. We're going to relabel this to a refactor since it's not an easy fix to actually address this problem.

Going forward, our proposed rules for replacements are as follows:

- If disabled
  - Does not copy over (disabled universally means no replacing)
  - Children that are not disabled are still copied over (so a parent can be used as a filter to replace children models)
- Match on Name
  - if types don't match, delete children (so function replacements work)
- If name does not exist in destination copy in (so new models can be added with replacements)

This will be worked on in a future sprint session.

**Comment by BrianCollinss** [comment]
Just a question: How about children that are disabled, but do not exist in the destination? My example shows that disabling such children does not really disable them.

**Comment by par456** [comment]
Yeah, at the moment the disabled flag is basically being ignored in replacements unless it's attached to a model directly under the replacements folder. We are hesitant to fix just that as it has a bunch of knock on effects to other replacement and edge cases that exist.

Under the new rules we've been putting together for this, a disabled node, whether it exists or not in the destination model, would not be included in when replacements does it's thing.

That is another edge case for us to be aware of though.
